templates/cluster-template-clusterclass.yaml

```diff
@@ -79,30 +79,6 @@ spec:
             kind: AzureMachineTemplate
             name: ${CLUSTER_NAME}-md-0
 ---
-apiVersion: cluster.x-k8s.io/v1beta1
```

Comment:
Is a breaking change for users relying on `clusterctl generate`?

Reply:
I moved the cluster topology to its own file since it is not recommended to define a Cluster topology and its ClusterClass at the same time. For users using `clusterctl generate`, does it require a `Cluster` object in a template to apply it to your cluster? If so, users will have to `kubectl apply -f` the ClusterClass separately but yeah it is a breaking change. @CecileRobertMichon wdyt?

Reply:
I don't think there are any requirements as to how we structure our templates, I can see where this might be frustrating if the script I had been using before with this template no longer actually produces a cluster.

Reply:
I do think this is a cleaner separation, and the likelihood that there is someone out there that gets broken by this change is pretty low. Hopefully, if you have scripts that rely on template files in a repo you are pinning the CAPZ version and testing the scripts before bumping the version...

so I'm leaning +1 on making this change